### ISSUE TITLE
Turn blacklist into whitelist for suggested apache config

### ISSUE THREAD

**Issue by eskildsen** [docker/docs#10589]
File: [registry/recipes/apache.md](https://docs.docker.com/registry/recipes/apache/)

_**tl;dr:** Update documentation recipe to match best practices. Users might use this for other projects as well._

Regarding the recipe for setting up a private docker registry using apache2 as reverse proxy, then you are advising to use authentication based on a blacklist for request methods:
```
# Read access to authentified users
<Limit GET HEAD>
    Require valid-user
</Limit>

# Write access to docker-deployer only
<Limit POST PUT DELETE PATCH>
    Require group pusher
</Limit>
```

This will only match the protocols listed above, but not other valid/invalid protocols. So if the client is not sending a request with any of above protocols, he will not be prompted for authentication. E.g. without any credentials we can bypass the access check completely and list the default alive endpoint:
```
$ curl https://domain.com/v2/ -X GETT -v
...
< HTTP/1.1 200 OK
< Date: Tue, 07 Apr 2020 10:41:32 GMT
< Server: Apache/2.4.25 (Debian)
< Content-Length: 2
< Content-Type: application/json; charset=utf-8
< Docker-Distribution-Api-Version: registry/2.0
< X-Content-Type-Options: nosniff
< 
{}
```

Depending on current/future endpoint configuration in the Docker Registry handlers it might be possible to leak information. Thus, please consider upgrading the referred documentation to something similar to this:
```
# Read access to authentified users
<LimitExcept POST PUT DELETE PATCH>
    Require valid-user
</LimitExcept>

# Write access to docker-deployer only
<Limit POST PUT DELETE PATCH>
    Require group pusher
</Limit>
```
